pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -178,6 +178,14 @@ private String getMultiStageQueryResponse(String query, String queryOptions, Htt
           new Exception("Unable to find table name from SQL thus cannot dispatch to broker.")).toString();
     }
 
+    for (String tableName : tableNames) {
```

Comment:
We are looping over these tables 3 times (once here, once in `getCommonBrokerTenant()`, once in `getCommonServerTenant()`), and check if table exists, which is one ZK access per check. Can we merge them into one method and only loop over the tables once?